pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,309 @@
+        values=[2e-3,10],
+        get_process= lambda v : float(v.split(" ",1)[-1][:-1]),
+        set_process=lambda v: "%.2eV"%v
+    )
+    coupling=Channel.control(
+        "C{ch}:CPL?","C{ch}:CPL %s1M",
+        "Sets and gets the channel coupling mode. (see UM p. 35)",
+        validator=truncated_discrete_set,
+        values={"DC":"D","AC":"A"},
+        map_values=True,
+        get_process= lambda v : v.split(" ",1)[-1][0],
+    )
+    def getWaveform(self):
+        """
+        Returns the waveforms displayed in the channel.
+        return:
+            time: (1d array) the time in seconds since the trigger epoch for every voltage value in the waveforms
+            voltages: (1d array) the waveform in V
+        """
+        command='C{ch}:WF? DAT2'
+        descriptorDictionnary=self.getDesc()
+        self.write(command) 
+        response=self.read_bytes(count=-1,break_on_termchar=True)
+        rawWaveform=list(struct.unpack_from('%db'%descriptorDictionnary["numDataPoints"],response,offset=descriptorDictionnary["descriptorOffset"]))
+        waveform=[point*descriptorDictionnary["verticalGain"]-descriptorDictionnary["verticalOffset"] for point in rawWaveform]
+        timetags=[i*descriptorDictionnary["horizInterval"]+descriptorDictionnary["horizOffset"] for i in range(len(rawWaveform))]
+        return timetags,waveform
+
+    def getDesc(self):
+        '''
+        Gets the descriptor of data being sent when querying device for waveform
+        :return:
+            dict: A dictionnary with the keys:
+                numDataPoints: the number of poitns in the waveform
+                verticalGain: the voltage increment per code value (in V)
+                verticalOffset: the voltage offset to add to the decoded voltage values
+                horizInterval: the time interval between points in s
+                horizOffset:the offset to add to the time steps
+                descriptorOffset: Length of the C1:WF ALL,#9000000346 message
+        '''
+        command='C{ch}:WF? DESC'
+        self.write(command)
+        descriptor=self.read_bytes(count=-1,break_on_termchar=True)
+        descriptorOffset=21
+        (numDataPoints,)=struct.unpack_from('l',descriptor,offset=descriptorOffset+60)
+        (verticalGain,)=struct.unpack_from('f',descriptor,offset=descriptorOffset+156)
+        (verticalOffset,)=struct.unpack_from('f',descriptor,offset=descriptorOffset+160)
+        (horizInterval,)=struct.unpack_from('f',descriptor,offset=descriptorOffset+176)
+        (horizOffset,)=struct.unpack_from('d',descriptor,offset=descriptorOffset+180)
+        descriptorDictionnary={
+            "numDataPoints":numDataPoints,
+            "verticalGain":verticalGain,
+            "verticalOffset":verticalOffset,
+            "horizInterval":horizInterval,
+            "horizOffset":horizOffset,
+            "descriptorOffset":descriptorOffset
+        }
+        return descriptorDictionnary
+class TriggerChannel(Channel):
```

Comment:
Two empty lines between definitions outside of classes, please.
```suggestion


class TriggerChannel(Channel):
```